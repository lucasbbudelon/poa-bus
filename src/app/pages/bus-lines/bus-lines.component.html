<section class="m-3">
  <h1>POA-Bus</h1>
  <p class="lead">Sistema integrado ao <a href="http://datapoa.com.br/group/about/mobilidade"
      target="_blanck">DataPoa</a>
    para consulta de linhas de ônibus e linhas de lotação.</p>

  <form [formGroup]="formGroup" *ngIf="data">

    <div class="form-group form-inline">
      Buscar por: <input class="form-control ml-2" type="text" formControlName="search" />
    </div>

    <table class="table table-striped">
      <thead>
        <tr>
          <th scope="col" style="width: 5%;">#</th>
          <th scope="col" style="width: 10%;">Linha</th>
          <th scope="col" style="width: 10%;">Tipo</th>
          <th scope="col" style="width: 45%;">Nome <i class="fas fa-sort-alpha-up"></i></th>
          <th scope="col" style="width: 15%;"></th>
        </tr>
      </thead>
      <tbody>
        <tr *ngFor="let busLine of data.items; let i = index">
          <th scope="row">
            {{ (i+1) + (currentPage > 1 ? (currentPage -1) * pageSize.value : 0) }}
          </th>
          <td>
            {{busLine.id}} # {{busLine.code}}
          </td>
          <td>
            <app-bus-lines-type [type]="busLine.type"></app-bus-lines-type>
          </td>
          <td>
            {{busLine.name}}
          </td>
          <td>
            <button type="button" class="btn btn-secondary" (click)="openItinerary(busLine)">
              <i class="fas fa-directions"></i> itinerário
            </button>
          </td>
        </tr>
      </tbody>
    </table>

    <div class="d-flex justify-content-between mt-2">
      <ngb-pagination 
        [collectionSize]="data.totalItems" 
        [(page)]="currentPage" 
        [pageSize]="pageSize.value"
        (pageChange)="loadNextPage()" 
        [maxSize]="paginationMaxSize" 
        [boundaryLinks]="false">
      </ngb-pagination>

      <select class="custom-select w-auto" formControlName="pageSize">
        <option *ngFor="let value of itemsPerPage" [ngValue]="value">
          {{value}} itens por página
        </option>
      </select>
    </div>
  </form>
</section>
