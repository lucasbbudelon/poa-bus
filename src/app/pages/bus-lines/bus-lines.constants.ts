import { NgbModalOptions } from '@ng-bootstrap/ng-bootstrap';

export const ITEMS_PER_PAGE = [5, 10, 15, 20, 25, 30];
export const ITEMS_PER_PAGE_DEFAULT = 10;
export const PAGINATION_MAX_SIZE = 10;
export const MODAL_ITINERARY_OPTIONS: NgbModalOptions = {
    scrollable: true,
    size: 'lg'
};
