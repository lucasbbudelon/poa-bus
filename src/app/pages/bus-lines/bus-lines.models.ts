export interface BusLine {
  id: string;
  code: string;
  name: string;
  type: BusLineType;
}

export interface BusLinePage {
  current?: number;
  limit?: number;
}

export interface BusLineData {
  page: BusLinePage;
  items: BusLine[];
  totalItems: number;
}

export enum BusLineType {
  Bus = 'o',
  Stocking = 'l',
}

export interface BusLineItinerary {
  busLine: BusLine;
  locations: Location[];
}

export interface BusLineItineraryLocation {
  order: number;
  lat: string;
  lng: string;
  urlMap: string;
}
