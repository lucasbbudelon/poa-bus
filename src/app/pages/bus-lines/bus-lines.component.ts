import { Component, OnDestroy, OnInit } from '@angular/core';
import { AbstractControl, FormControl, FormGroup } from '@angular/forms';
import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
import { Subscription } from 'rxjs';
import { finalize, switchMap, tap } from 'rxjs/operators';

import { ITEMS_PER_PAGE, ITEMS_PER_PAGE_DEFAULT, MODAL_ITINERARY_OPTIONS, PAGINATION_MAX_SIZE } from './bus-lines.constants';
import { BusLine, BusLineData, BusLinePage } from './bus-lines.models';
import { BusLinesService } from './bus-lines.service';
import { BusLinesItineraryComponent } from './components/bus-lines-itinerary/bus-lines-itinerary.component';

@Component({
  selector: 'app-bus-lines',
  templateUrl: './bus-lines.component.html',
  styleUrls: ['./bus-lines.component.scss'],
})
export class BusLinesComponent implements OnInit, OnDestroy {
  get search(): AbstractControl {
    return this.formGroup.get('search');
  }
  get pageSize(): AbstractControl {
    return this.formGroup.get('pageSize');
  }
  get itemsPerPage() {
    return ITEMS_PER_PAGE;
  }
  get paginationMaxSize() {
    return PAGINATION_MAX_SIZE;
  }
  get firstPage(): BusLinePage {
    return { current: 1, limit: this.pageSize.value };
  }
  get nextPage(): BusLinePage {
    return { current: this.currentPage++, limit: this.pageSize.value };
  }

  public formGroup: FormGroup;
  public data: BusLineData;
  public currentPage: number;

  private subscriptions: Subscription;

  constructor(
    private busLinesService: BusLinesService,
    private modalService: NgbModal
  ) { }

  ngOnInit() {
    this.buildForm();
    this.loadFirstPage();
  }

  ngOnDestroy() {
    if (this.subscriptions && !this.subscriptions.closed) {
      this.subscriptions.unsubscribe();
    }
  }

  buildForm() {
    this.formGroup = new FormGroup({
      search: new FormControl(null),
      pageSize: new FormControl(ITEMS_PER_PAGE_DEFAULT),
    });

    this.subscriptions = new Subscription();
    this.subscribeToSearchChanges();
    this.subscribeToPageSizeChanges();
  }

  subscribeToSearchChanges() {
    const searchValueChanges$ = this.search.valueChanges
      .pipe(
        switchMap((filter) => this.busLinesService
          .getBusLines(this.firstPage, filter))
      )
      .subscribe((data) => this.data = data);

    this.subscriptions.add(searchValueChanges$);
  }

  subscribeToPageSizeChanges() {
    const pageSizeValueChanges$ = this.pageSize.valueChanges
      .pipe(
        switchMap((value) => this.busLinesService
          .getBusLines({ current: this.currentPage, limit: value }))
      )
      .subscribe((data) => this.data = data);

    this.subscriptions.add(pageSizeValueChanges$);
  }

  loadFirstPage() {
    this.busLinesService
      .getBusLines(this.firstPage)
      .pipe(
        tap((data) => this.data = data),
        tap((data) => this.currentPage = data.page.current),
      )
      .subscribe();
  }

  loadNextPage() {
    this.busLinesService
      .getBusLines(this.nextPage)
      .pipe(
        tap((data) => this.data = data),
        tap((data) => this.currentPage = data.page.current),
      )
      .subscribe();
  }

  openItinerary(busLine: BusLine) {
    this.busLinesService
      .getItinerary(busLine)
      .subscribe((itinerary) => {
        const busLinesItineraryComponent = this.modalService.open(BusLinesItineraryComponent, MODAL_ITINERARY_OPTIONS);
        busLinesItineraryComponent.componentInstance.itinerary = itinerary;
      });
  }
}
