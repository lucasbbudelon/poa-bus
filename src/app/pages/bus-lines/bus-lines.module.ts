import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { NgbAccordionModule, NgbPaginationModule } from '@ng-bootstrap/ng-bootstrap';

import { BusLinesComponent } from './bus-lines.component';
import { BusLinesRouting } from './bus-lines.routing';
import { BusLinesItineraryComponent } from './components/bus-lines-itinerary/bus-lines-itinerary.component';
import { BusLinesTypeComponent } from './components/bus-lines-type/bus-lines-type.component';

@NgModule({
  declarations: [
    BusLinesComponent,
    BusLinesItineraryComponent,
    BusLinesTypeComponent
  ],
  imports: [
    CommonModule,
    BusLinesRouting,
    FormsModule,
    ReactiveFormsModule,
    NgbPaginationModule,
    NgbAccordionModule
  ],
})
export class BusLinesModule { }
