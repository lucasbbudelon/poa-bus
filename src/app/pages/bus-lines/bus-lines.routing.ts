import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';

import { BusLinesComponent } from './bus-lines.component';

const routes: Routes = [
  {
    path: '',
    component: BusLinesComponent,
  },
];

@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule],
})
export class BusLinesRouting {}
