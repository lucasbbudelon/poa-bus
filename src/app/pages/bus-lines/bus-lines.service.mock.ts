import { of } from 'rxjs';

import { BusLine, BusLineData, BusLineItinerary, BusLineType } from './bus-lines.models';

export class BusLinesServiceMock {
    getBusLines = jest.fn().mockReturnValue(of(BUS_LINE_DATA_MOCK));
    getItinerary = jest.fn().mockReturnValue(of(BUS_LINES_ITINERARY_MOCK));
}

export const BUS_LINE_MOCK: BusLine = {
    id: '5485',
    code: 'T11-1',
    name: '3ª PERIMETRAL',
    type: BusLineType.Bus
};

export const BUS_LINE_DATA_MOCK: BusLineData = {
    page: {
        limit: 10,
        current: 1
    },
    items: [
        {
            id: '5485',
            code: 'T11-1',
            name: '3ª PERIMETRAL',
            type: BusLineType.Bus
        },
        {
            id: '5978',
            code: '7052-1',
            name: 'AEROPORTO/CEASA',
            type: BusLineType.Bus
        },
        {
            id: '5414',
            code: '282-2',
            name: 'CRUZEIRO DO SUL',
            type: BusLineType.Bus
        },
        {
            id: '5745',
            code: '274-2',
            name: 'GLORIA/AZENHA/CASCATINHA',
            type: BusLineType.Bus
        },
        {
            id: '52',
            code: '50.11-1',
            name: 'AUXILIADORA - ANITA',
            type: BusLineType.Stocking
        },
        {
            id: '15',
            code: '10.4-1',
            name: 'IPANEMA',
            type: BusLineType.Stocking
        },
    ],
    totalItems: 6
};

export const BUS_LINES_ITINERARY_MOCK: BusLineItinerary = {
    busLine: this.BUS_LINE_MOCK,
    locations: []
};

export const BUS_LINE_API_MOCK = [
    {
        id: '5665',
        codigo: '490-2',
        nome: 'MORRO SANTANA'
    },
    {
        id: '5486',
        codigo: 'T11-2',
        nome: '3ª PERIMETRAL'
    },
    {
        id: '5726',
        codigo: '178-2',
        nome: 'PRAIA DE BELAS'
    },
    {
        id: '5833',
        codigo: '855-1',
        nome: 'PROTASIO/CAIRU'
    },
    {
        id: '5612',
        codigo: '353-1',
        nome: 'IPIRANGA / PUC / UFRGS'
    },
    {
        id: '5583',
        codigo: '340-1',
        nome: 'JARDIM BOTANICO'
    },
    {
        id: '5024',
        codigo: '184-1',
        nome: 'JUCA BATISTA'
    },
    {
        id: '5680',
        codigo: '4951-1',
        nome: 'MANOEL ELIAS/MORRO SANTANA'
    },
    {
        id: '32',
        codigo: '20.6-2',
        nome: 'GLORIA'
    },
    {
        id: '45',
        codigo: '40.2-1',
        nome: 'JOAO ABOTT'
    },
];

export const BUS_LINES_ITINERARY_API_MOCK = {
    0: {
        lat: '-30.02091857422600000',
        lng: '-51.10401713620700000'
    },
    1: {
        lat: '-30.02080957422600000',
        lng: '-51.10340913620700000'
    },
    2: {
        lat: '-30.02079457422600000',
        lng: '-51.10332613620700000'
    },
    3: {
        lat: '-30.02075357422600000',
        lng: '-51.10310013620700000'
    },
    4: {
        lat: '-30.02071757422600000',
        lng: '-51.10289913620700000'
    },
    5: {
        lat: '-30.02092657422600000',
        lng: '-51.10284513620700000'
    },
    6: {
        lat: '-30.02152757422600000',
        lng: '-51.10274113620700000'
    },
    7: {
        lat: '-30.02172157422600000',
        lng: '-51.10270213620700000'
    },
    8: {
        lat: '-30.02215557422600000',
        lng: '-51.10262813620700000'
    },
    9: {
        lat: '-30.02238257422600000',
        lng: '-51.10259513620700000'
    },
    idlinha: BUS_LINE_MOCK.id,
    nome: BUS_LINE_MOCK.name,
    codigo: BUS_LINE_MOCK.code
};
