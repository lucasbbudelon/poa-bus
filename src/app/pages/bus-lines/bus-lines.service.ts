import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
import { forkJoin, Observable, of } from 'rxjs';
import { map, tap, finalize } from 'rxjs/operators';
import { environment } from 'src/environments/environment';

import { ITEMS_PER_PAGE_DEFAULT } from './bus-lines.constants';
import {
  BusLine,
  BusLineData,
  BusLineItinerary,
  BusLineItineraryLocation,
  BusLinePage,
  BusLineType,
} from './bus-lines.models';

@Injectable({
  providedIn: 'root',
})
export class BusLinesService {

  busLines: BusLine[];

  constructor(private httpClient: HttpClient, private sanitizer: DomSanitizer) { }

  getBusLines(page: BusLinePage, filter?: string): Observable<BusLineData> {

    if (this.busLines) {
      const busLines = filter ? this.filterBusLinesByAnyValue(filter) : this.busLines;
      return of(this.loadBusLinesData(busLines, page));
    }

    const getBusLines = this.httpClient
      .get<any[]>(`${environment.apiDataPoa}/php/facades/process.php?a=nc&p=%25&t=${BusLineType.Bus}`)
      .pipe(
        map((busLines) => busLines.map((item) => this.mapBusLine(item, BusLineType.Bus))),
      );

    const getStockingLines = this.httpClient
      .get<any[]>(`${environment.apiDataPoa}/php/facades/process.php?a=nc&p=%25&t=${BusLineType.Stocking}`)
      .pipe(
        map((busLines) => busLines.map((item) => this.mapBusLine(item, BusLineType.Stocking))),
      );

    return forkJoin([getBusLines, getStockingLines])
      .pipe(
        map(([busLines, stockingLines]) => busLines.concat(stockingLines)),
        tap((busLines) => this.busLines = this.sortBusLinesByName(busLines)),
        map((busLines) => this.loadBusLinesData(busLines, page)),
      );
  }

  getItinerary(busLine: BusLine): Observable<BusLineItinerary> {
    return this.httpClient
      .get<any>(`${environment.apiDataPoa}/php/facades/process.php?a=il&p=${busLine.id}`)
      .pipe(
        map((itinerary) => this.mapItinerary(itinerary, busLine.type)),
      );
  }

  private mapBusLine(modelApi: any, type: BusLineType): BusLine {
    return {
      id: modelApi.id || modelApi.idlinha,
      code: modelApi.codigo,
      name: modelApi.nome,
      type
    };
  }

  private mapItinerary(modelApi: any, type: BusLineType): BusLineItinerary {
    return {
      busLine: this.mapBusLine(modelApi, type),
      locations: this.mapLocations(modelApi)
    };
  }

  private mapLocations(modelApi: any): Location[] {
    const locations = Object
      .keys(modelApi)
      .filter((locationIndex) => locationIndex.match(/^[0-9]+$/) != null);

    return locations
      .map((locationIndex) => ({
        ...modelApi[locationIndex],
        order: parseInt(locationIndex, 10),
        urlMap: this.getUrlMapLocation(modelApi[locationIndex])
      }));
  }

  private loadBusLinesData(busLines: BusLine[], page: BusLinePage): BusLineData {

    const size = (page.current - 1) * page.limit;

    return {
      page,
      items: busLines.slice(size, size + page.limit),
      totalItems: busLines.length
    };
  }

  private sortBusLinesByName(busLines: BusLine[]): BusLine[] {
    return busLines.sort((a, b) => {
      if (a.name > b.name) {
        return 1;
      }
      if (a.name < b.name) {
        return -1;
      }
      return 0;
    });
  }

  private filterBusLinesByAnyValue(term: string) {
    term = term.toLowerCase();
    return this.busLines
      .filter((busLine) =>
        busLine.id.toLowerCase().includes(term) ||
        busLine.code.toLowerCase().includes(term) ||
        busLine.name.toLowerCase().includes(term)
      );
  }

  private getUrlMapLocation(location: BusLineItineraryLocation) {
    const q = `q=${location.lat},${location.lng}`;
    const zoom = 'z=20';
    const output = 'output=embed';
    const url = `${environment.apiGoogleMaps}/maps?${q}&${zoom}&${output}`;
    return this.sanitizer.bypassSecurityTrustResourceUrl(url);
  }
}
