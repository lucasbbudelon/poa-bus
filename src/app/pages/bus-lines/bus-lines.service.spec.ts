import { of } from 'rxjs';
import { DomSanitizerMock } from 'src/app/test/mocks/dom-sanitizer.mock';
import { HttpClientMock } from 'src/app/test/mocks/http-client.mock';
import { environment } from 'src/environments/environment';

import { PAGINATION_MAX_SIZE } from './bus-lines.constants';
import { BusLineType } from './bus-lines.models';
import { BusLinesService } from './bus-lines.service';
import {
  BUS_LINE_API_MOCK,
  BUS_LINE_DATA_MOCK,
  BUS_LINE_MOCK,
  BUS_LINES_ITINERARY_API_MOCK,
} from './bus-lines.service.mock';

describe('BusLineService', () => {
  let service: BusLinesService;
  let sanitizer: DomSanitizerMock;
  let httpMock: HttpClientMock;

  beforeEach(() => {
    httpMock = new HttpClientMock();
    sanitizer = new DomSanitizerMock();
    service = new BusLinesService(httpMock as any, sanitizer as any);
    jest.clearAllMocks();
  });

  it('should be created', () => {
    expect(service).toBeTruthy();
  });

  describe('when fetching bus lines', () => {

    beforeEach(() => {
      httpMock.get.mockReturnValue(of(BUS_LINE_API_MOCK));
    });

    describe('should data is not loaded', () => {

      beforeEach(() => {
        Object.defineProperty(service, 'busLines', {
          get: jest.fn().mockReturnValue(null),
        });
      });

      it('should call api data poa', () => {
        service
          .getBusLines({ current: 1, limit: PAGINATION_MAX_SIZE })
          .subscribe((data) => {
            expect(httpMock.get).toHaveBeenCalledWith(`${environment.apiDataPoa}/php/facades/process.php?a=nc&p=%25&t=${BusLineType.Bus}`);
            expect(httpMock.get).toHaveBeenCalledWith(`${environment.apiDataPoa}/php/facades/process.php?a=nc&p=%25&t=${BusLineType.Stocking}`);
          });
      });
    });

    describe('should data is already loaded', () => {

      beforeEach(() => {
        Object.defineProperty(service, 'busLines', {
          get: jest.fn().mockReturnValue(BUS_LINE_DATA_MOCK.items),
        });
      });

      it('don"t should call api data poa', () => {
        service
          .getBusLines({ current: 1, limit: PAGINATION_MAX_SIZE })
          .subscribe((data) => {
            expect(httpMock.get).not.toHaveBeenCalled();
            expect(httpMock.get).not.toHaveBeenCalled();
          });
      });
    });

    describe('should return paged results', () => {
      it('page one', () => {
        service
          .getBusLines({ current: 1, limit: 5 })
          .subscribe((data) => expect(data).toMatchSnapshot());
      });

      it('page two', () => {
        service
          .getBusLines({ current: 2, limit: 5 })
          .subscribe((data) => expect(data).toMatchSnapshot());
      });

      it('page three', () => {
        service
          .getBusLines({ current: 3, limit: 5 })
          .subscribe((data) => expect(data).toMatchSnapshot());
      });
    });

    describe('when filter values', () => {

      beforeEach(() => {
        Object.defineProperty(service, 'busLines', {
          get: jest.fn().mockReturnValue(BUS_LINE_DATA_MOCK.items),
        });
      });

      it('should filter by id', () => {
        service
          .getBusLines({ current: 1, limit: PAGINATION_MAX_SIZE }, BUS_LINE_DATA_MOCK.items[2].id)
          .subscribe((data) => expect(data).toMatchSnapshot());
      });

      it('should filter by code', () => {
        service
          .getBusLines({ current: 1, limit: PAGINATION_MAX_SIZE }, BUS_LINE_DATA_MOCK.items[1].code)
          .subscribe((data) => expect(data).toMatchSnapshot());
      });

      it('should filter by name', () => {
        service
          .getBusLines({ current: 1, limit: PAGINATION_MAX_SIZE }, BUS_LINE_DATA_MOCK.items[4].name)
          .subscribe((data) => expect(data).toMatchSnapshot());
      });
    });
  });

  it('when fetching itinerary', () => {

    httpMock.get.mockReturnValue(of(BUS_LINES_ITINERARY_API_MOCK));

    service
      .getItinerary(BUS_LINE_MOCK)
      .subscribe((data) => {
        expect(httpMock.get).toHaveBeenCalledWith(`${environment.apiDataPoa}/php/facades/process.php?a=il&p=${BUS_LINE_MOCK.id}`);
        expect(data).toMatchSnapshot();
      });
  });
});
