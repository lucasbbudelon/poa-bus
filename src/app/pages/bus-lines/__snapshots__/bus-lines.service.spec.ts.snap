// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusLineService when fetching bus lines should return paged results page one 1`] = `
Object {
  "items": Array [
    Object {
      "code": "T11-2",
      "id": "5486",
      "name": "3ª PERIMETRAL",
      "type": "o",
    },
    Object {
      "code": "T11-2",
      "id": "5486",
      "name": "3ª PERIMETRAL",
      "type": "l",
    },
    Object {
      "code": "20.6-2",
      "id": "32",
      "name": "GLORIA",
      "type": "o",
    },
    Object {
      "code": "20.6-2",
      "id": "32",
      "name": "GLORIA",
      "type": "l",
    },
    Object {
      "code": "353-1",
      "id": "5612",
      "name": "IPIRANGA / PUC / UFRGS",
      "type": "o",
    },
  ],
  "page": Object {
    "current": 1,
    "limit": 5,
  },
  "totalItems": 20,
}
`;

exports[`BusLineService when fetching bus lines should return paged results page three 1`] = `
Object {
  "items": Array [
    Object {
      "code": "184-1",
      "id": "5024",
      "name": "JUCA BATISTA",
      "type": "o",
    },
    Object {
      "code": "184-1",
      "id": "5024",
      "name": "JUCA BATISTA",
      "type": "l",
    },
    Object {
      "code": "4951-1",
      "id": "5680",
      "name": "MANOEL ELIAS/MORRO SANTANA",
      "type": "o",
    },
    Object {
      "code": "4951-1",
      "id": "5680",
      "name": "MANOEL ELIAS/MORRO SANTANA",
      "type": "l",
    },
    Object {
      "code": "490-2",
      "id": "5665",
      "name": "MORRO SANTANA",
      "type": "o",
    },
  ],
  "page": Object {
    "current": 3,
    "limit": 5,
  },
  "totalItems": 20,
}
`;

exports[`BusLineService when fetching bus lines should return paged results page two 1`] = `
Object {
  "items": Array [
    Object {
      "code": "353-1",
      "id": "5612",
      "name": "IPIRANGA / PUC / UFRGS",
      "type": "l",
    },
    Object {
      "code": "340-1",
      "id": "5583",
      "name": "JARDIM BOTANICO",
      "type": "o",
    },
    Object {
      "code": "340-1",
      "id": "5583",
      "name": "JARDIM BOTANICO",
      "type": "l",
    },
    Object {
      "code": "40.2-1",
      "id": "45",
      "name": "JOAO ABOTT",
      "type": "o",
    },
    Object {
      "code": "40.2-1",
      "id": "45",
      "name": "JOAO ABOTT",
      "type": "l",
    },
  ],
  "page": Object {
    "current": 2,
    "limit": 5,
  },
  "totalItems": 20,
}
`;

exports[`BusLineService when fetching bus lines when filter values should filter by code 1`] = `
Object {
  "items": Array [
    Object {
      "code": "7052-1",
      "id": "5978",
      "name": "AEROPORTO/CEASA",
      "type": "o",
    },
  ],
  "page": Object {
    "current": 1,
    "limit": 10,
  },
  "totalItems": 1,
}
`;

exports[`BusLineService when fetching bus lines when filter values should filter by id 1`] = `
Object {
  "items": Array [
    Object {
      "code": "282-2",
      "id": "5414",
      "name": "CRUZEIRO DO SUL",
      "type": "o",
    },
  ],
  "page": Object {
    "current": 1,
    "limit": 10,
  },
  "totalItems": 1,
}
`;

exports[`BusLineService when fetching bus lines when filter values should filter by name 1`] = `
Object {
  "items": Array [
    Object {
      "code": "50.11-1",
      "id": "52",
      "name": "AUXILIADORA - ANITA",
      "type": "l",
    },
  ],
  "page": Object {
    "current": 1,
    "limit": 10,
  },
  "totalItems": 1,
}
`;

exports[`BusLineService when fetching itinerary 1`] = `
Object {
  "busLine": Object {
    "code": "T11-1",
    "id": "5485",
    "name": "3ª PERIMETRAL",
    "type": "o",
  },
  "locations": Array [
    Object {
      "lat": "-30.02091857422600000",
      "lng": "-51.10401713620700000",
      "order": 0,
      "urlMap": "safeResourceUrl",
    },
    Object {
      "lat": "-30.02080957422600000",
      "lng": "-51.10340913620700000",
      "order": 1,
      "urlMap": "safeResourceUrl",
    },
    Object {
      "lat": "-30.02079457422600000",
      "lng": "-51.10332613620700000",
      "order": 2,
      "urlMap": "safeResourceUrl",
    },
    Object {
      "lat": "-30.02075357422600000",
      "lng": "-51.10310013620700000",
      "order": 3,
      "urlMap": "safeResourceUrl",
    },
    Object {
      "lat": "-30.02071757422600000",
      "lng": "-51.10289913620700000",
      "order": 4,
      "urlMap": "safeResourceUrl",
    },
    Object {
      "lat": "-30.02092657422600000",
      "lng": "-51.10284513620700000",
      "order": 5,
      "urlMap": "safeResourceUrl",
    },
    Object {
      "lat": "-30.02152757422600000",
      "lng": "-51.10274113620700000",
      "order": 6,
      "urlMap": "safeResourceUrl",
    },
    Object {
      "lat": "-30.02172157422600000",
      "lng": "-51.10270213620700000",
      "order": 7,
      "urlMap": "safeResourceUrl",
    },
    Object {
      "lat": "-30.02215557422600000",
      "lng": "-51.10262813620700000",
      "order": 8,
      "urlMap": "safeResourceUrl",
    },
    Object {
      "lat": "-30.02238257422600000",
      "lng": "-51.10259513620700000",
      "order": 9,
      "urlMap": "safeResourceUrl",
    },
  ],
}
`;
